src/Donations/LeftPanel/index.tsx

```diff
@@ -26,10 +27,20 @@ function LeftPanel(): ReactElement {
     callbackUrl,
   } = useContext(QueryParamContext);
 
-  const [isMobile, setIsMobile] = useState(false);
+  const router = useRouter();
+  const [showCancelButton, setShowCancelButton] = useState(false);
+
+  useEffect(() => {
+    if (!router.query.step) {
+      setShowCancelButton(true);
+    }
+  }, [router]);
 
```

Comment:
1. Avoid confusion between the `showCancelButton` state variable, and the `canShowBackButton` / `canShowCancelButton` variable. Perhaps `showCancelButton` should be renamed to something that indicates what it represents better. 
2. Checking `step` as a proxy to decide which page the user came on won't work if the user comes directly to the step 1 page (the link usually has `to` and `step` is implicitly understood). Also, the cancel button will disappear if the user refreshes the page after clicking on a project from `Select Project`. Don't depend on state for this information.